### DISCUSSION TITLE
CLI Command to run the Search ReIndex

### DISCUSSION THREAD

**Discussion by CreativePoweredGeek** [ExpressionEngine/ExpressionEngine#3428] [Feature Requests]
### Suggest an idea for ExpressionEngine

**Is your feature request related to a problem? Please describe.**
When Running the Search Reindex on a Large number of Entries, It will frequently hit the Timeouts. I currently have a DB with over 33,000 entries to reindex and it just can't manage it. It would be good as well, if when refreshing the index for a collection in ProSearch, it did this as well.

**Describe the solution you'd like**
Just a CLI to run this so as not to hit the Timeouts

**Describe alternatives you've considered**
--

**Teachability, Documentation, Adoption, Migration Strategy**
-- 

- [ ] I am capable and would like to work on implementation of this feature if it is considered.

**Comment by intoeetive** [comment]
@CreativePoweredGeek this is excellent idea. If you have time and desire to work on it, please go ahead. Thank you!

**Comment by CreativePoweredGeek** [comment]
@intoeetive  to be 100% honest... I wouldn't even know where to start on this.  CLI is totally new to me.  This is a result of an issue I have and discussed on Slack. Tom suggested I put it in as a feature request.

**Reply by intoeetive** [reply]
@CreativePoweredGeek no worries - I just misinterpreted the fact that you left the " I am capable and would like to work..." untouched.

We'll get this sorted out

**Comment by intoeetive** [comment]
We'll get this added with https://github.com/ExpressionEngine/ExpressionEngine/pull/3553

**Reply by CreativePoweredGeek** [reply]
That would be great.  Thank you.   Just to toss a thought into this... it would be great to be able to decide which channels you want to reindex.   I have a site right now that has a lot of grid fields but only one channel really needs to be reindexed.  It seems like a waste to have to spend the resources to reindex channels that typically wouldn't need to be reindexed.  Anyway... just adding to the wishlist. :)
